### DISCUSSION TITLE
End of stream reached before reading the desired number of bytes when Demo.Start(File.OpenRead(path))

### DISCUSSION THREAD

**Discussion by elkraps** [saul/demofile-net#8] [Q&A] [answered]
Hi, im trying to parse demo from cs2. I checked my file is not broken or empty. Also there are no other processes using that file. Have you got any ideas why i get this error "End of stream reached before reading the desired number of bytes"? Thanks in advance :)
```
public async Task<string> ProcessLog(string path) {

    var error = string.Empty;
    var demo = new DemoParser();

    demo.Source1GameEvents.GameEnd += async e =>
    { 
        foreach(var player in demo.TeamCounterTerrorist.CSPlayerControllers) { 
            error += await this.UpdatePlayerStat(player); 
        }
        foreach(var player in demo.TeamTerrorist.CSPlayerControllers) {
            error += await this.UpdatePlayerStat(player);
        }  
    };

    try {
        var bytes = File.OpenRead(path);
        await demo.Start(bytes);
    } catch(Exception ex) {
        error = "An exception occurred: " + ex.Message;
    }
    return error;
}
```

**Comment by saul** [comment]
Please can you upload the demo?

**Reply by elkraps** [reply]
[auto-20231107-1955-de_mirage-Counter-Strike_2.zip](https://github.com/saul/demofile-net/files/13284855/auto-20231107-1955-de_mirage-Counter-Strike_2.zip)

**Reply by saul** [reply]
That hasn't uploaded properly - you might need to wait until the upload it complete before commenting.

**Reply by elkraps** [reply]
Sorry, updated

**Comment by saul** [comment, marked as answer]
I've just taken a look at the demo and a couple of things stand out:

1. This is a POV/in-eye demo, and these aren't supported just yet (I don't think any CS2 parsers handle these properly at the moment)
2. The demo stream seems to abruptly end - just before the demo should contain 33KB packet, the file ends. That's why you see this exception. I will likely change the exception to be something you could easily catch, like a `DemoEndedEarlyException`. Unfortunately we saw these a lot in CSGO as well, where the demo file didn't seem to have been fully written.

Hope this answers your question. The main issue is that the demo wasn't written to disk fully.

**Reply by elkraps** [reply]
Thanks a lot! 
Figured out that problem with incomplete file was because we need to intentionally start and stop demo with commands(tv_record [demoname]/tv_stoprecord). Command "tv_autorecord 1" doesnot work properly for me. 
PS. To enable gotv record i had to change map after "tv_enable" (for newbies :) )
